### PR TITLE
Add some tests to catch memory and resource links

### PR DESCRIPTION
**Pull request by jeskew**
/cc @mtdowling

### CODE REVIEW

File: features/performance/streaming.feature [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    And I take a snapshot of my resources
+    When I upload the file
+    Then I should not have leaked any resources
+
+    Examples:
+    | bytes       |
+    | 2048        |
+    | 2097152     |
+    | 20971520    |
+    | 2147483648  |
```
Comment (mtdowling) [inline features/performance/streaming.feature]: Do these examples actually cause us to try with a 2gb file?

Comment (jeskew) [inline features/performance/streaming.feature]: Absolutely!

Comment (jeskew) [inline features/performance/streaming.feature]: I'll drop it to 200MB.

**Comment by mtdowling** [comment]
:ship:

**Pull request merged**